dotCMS/src/integration-test/java/com/dotmarketing/portlets/fileassets/business/FileAssetAPITest.java

```diff
@@ -168,6 +170,54 @@ public void Test_Rename_File_Asset_Then_Recover_File_Then_Expect_Match()
 
     }
 
-
+  /**
+   * file assets should be stored in cache once they've been hydrated. This method tests to make sure
+   * that our cache is returning a fileAsset if it has been stored as one
+   * 
+   * @throws Exception
+   */
+    @Test
+    public void Test_That_File_Asset_Gets_Stored_in_Cache_and_is_Not_Rebuilt_Everytime()
+        throws Exception {
+
+      final Folder parentFolder = new FolderDataGen().nextPersisted();
+
+      // Lets make up some fictional data.
+
+      final java.io.File file = java.io.File.createTempFile("blah", ".txt");
+      FileUtil.write(file, "helloworld");
+      
+      final FileAssetDataGen fileAssetDataGen = new FileAssetDataGen(parentFolder, file);
+      final Contentlet con = fileAssetDataGen.nextPersisted();
+  
+      
+      // content from the content API will be content, first hit
+      Contentlet con1 = APILocator.getContentletAPI().find(con.getInode(), APILocator.systemUser(), false);
+      assertTrue("Content should not be a file asset", (!(con1 instanceof FileAsset)));
+      
+      // content from the content cache will be content, first hit
+      Contentlet con2 = CacheLocator.getContentletCache().get(con.getInode());
+      assertTrue("Contentlet from find comes from cache", con1==con2);
+      
+      // if you pipe that through the fileAssetAPI, it becomes a new file asset
+      FileAsset asset  = APILocator.getFileAssetAPI().find(con.getInode(), APILocator.systemUser(), false);
+      assertTrue("FileAsset should be a file asset", (asset instanceof FileAsset));
+      
+      // it is not just a mutated version of the contentlet
+      assertTrue("FileAsset is a new object, not a mutated contentlet", (asset != con2));
+      
+      
+      FileAsset asset2  = APILocator.getFileAssetAPI().find(con.getInode(), APILocator.systemUser(), false);
+      assertTrue("FileAssets should be the same Object", asset == asset2);
+      
+      
+      FileAsset asset3  = APILocator.getFileAssetAPI().fromContentlet(asset2);
+      assertTrue("FileAssets should be the same Object", asset3 == asset2);
+      
+      Contentlet asset4 = APILocator.getContentletAPI().find(con.getInode(), APILocator.systemUser(), false);
+      assertTrue("Content should not be a file asset", (asset4 instanceof FileAsset));
+      assertTrue("FileAssets should be the same Object", asset3 == asset4);
```

Comment:
This test looks perfect, but I am missing where the magic is happening, where we put the newly transformed fileAsset back into the contentlet cache?